x-pack/plugins/stack_connectors/public/connector_types/email/email_params.tsx

```diff
@@ -18,6 +18,22 @@ import { EmailActionParams } from '../types';
 
 const noop = () => {};
 
+export const getFormattedEmailOptions = (
+  searchValue: string,
+  previousOptions: Array<{ label: string }>
+): Array<{ label: string }> => {
+  if (!searchValue.trim()) return previousOptions;
+  const previousEmails: string[] = previousOptions.map((option) => option.label);
+  const newEmails: string[] = searchValue.split(',').reduce((result, email) => {
```

Comment:
I see no extra value using reduce here. You should replace it for a `[].map().filter(Boolean)` pattern as it's more intuitive and clearer. Although if you follow my next advice, the "using a Set from the beginning one" you'll have to use forEach instead

Reply:
Thank you for the detailed review! I've implemented the requested changes, including using a Set from the start as you suggested.